### ISSUE TITLE
[napari-workflows] How to rescale some (which?) labeling parameters when `level>0`?

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#152]
As part of https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/148, labeling workflows can be run at arbitrary resolution levels. This, however, means that the original parameters in the YAML file may be wrong, depending on the level we choose. I provide an example below (the relevant parameters are [`spot_sigma` and `outline_sigma`](https://github.com/haesleinhuepf/napari-segment-blobs-and-things-with-membranes/blob/main/napari_segment_blobs_and_things_with_membranes/__init__.py#L500-L504)).

In `cellpose_segmentation`, we were handling this ourselves by rescaling the input diameter when working at `level>0`. In napari-workflows we don't have an easy way to do so (I guess there is a way to do it, modifying `napari-workflows` objects, but this will require some more information about which parameters are to be rescaled, and by which factor..).

Any ideas on how to proceed?

Possible ways out:
1. We rely on the user only: if they want to run a workflow with `level>0`, then they should update the parameters in the YAML file accordingly. Question: can we enforce it? I guess not, and that we can only issue some warnings.
2. We modify the `napari-workflows` objects by hand, within the wrapper, if `level>0`. Question: how do we know which parameters are to be rescaled and by which factor?

### Example
Original workflow file:
```yaml
!!python/object:napari_workflows._workflow.Workflow
_tasks:
  Result of Expand labels (scikit-image, nsbatwm): !!python/tuple
  - !!python/name:napari_segment_blobs_and_things_with_membranes.expand_labels ''
  - Result of Voronoi-Otsu-labeling (nsbatwm)
  - 2.0
  Result of Median (scipy, nsbatwm): !!python/tuple
  - !!python/name:napari_segment_blobs_and_things_with_membranes.median_filter ''
  - input
  - 2.0
  Result of Voronoi-Otsu-labeling (nsbatwm): !!python/tuple
  - !!python/name:napari_segment_blobs_and_things_with_membranes.voronoi_otsu_labeling ''
  - Result of Median (scipy, nsbatwm)
  - 15.0
  - 2.0
```

Labeling at `level=2`:
![image](https://user-images.githubusercontent.com/3862206/198039920-1081463c-371b-497c-8056-ad68be8d7981.png)

**Comment by jluethi** [comment]
Good question. As a start, I would say this is the user's business though. The default in general is that everything is run at level 0. When would things be run at level 1, 2 etc? Mostly if one does segmentation, especially with neural networks. Users should adapt their parameters of napari workflows to the level they want to run workflows at.

I'm working on a ROI loading plugin that should allow users to load a specific ROI into napari to make it easy to build napari workflows on small subsets of the data. As part of that, I'll include a `level` parameter to allow users to load the data at a lower resolution level.

Thus, if a user builds a napari workflow using a lower resolution level image, they should then also run it at the resolution level through Fractal. But I don't think Fractal should worry about adapting the content of the pipelines.

**Comment by tcompa** [comment]
> Good question. As a start, I would say this is the user's business though. The default in general is that everything is run at level 0. When would things be run at level 1, 2 etc? Mostly if one does segmentation, especially with neural networks. Users should adapt their parameters of napari workflows to the level they want to run workflows at.

Yes, this was only meant to concern labeling tasks.

> I'm working on a ROI loading plugin that should allow users to load a specific ROI into napari to make it easy to build napari workflows on small subsets of the data. As part of that, I'll include a `level` parameter to allow users to load the data at a lower resolution level.

That will be interesting, also for our tests. I'm curious to see how it looks.


> Thus, if a user builds a napari workflow using a lower resolution level image, they should then also run it at the resolution level through Fractal. But I don't think Fractal should worry about adapting the content of the pipelines.

TL;DR: it's the user's business, we won't change anything (and I agree with this).

Closing. Please re-open if you have an idea for a reasonable warning to raise.
